src/app/modules/reports/components/time-entries-table/time-entries-table.component.spec.ts

```diff
@@ -76,6 +79,24 @@ describe('Reports Page', () => {
       expect(component.dtTrigger.next).toHaveBeenCalled();
     });
 
+    it('when the uri starts with http or https it should return true and open the url in a new tab', () => {
+      const uriExpected = 'http://customuri.com';
+      spyOn(component, 'isValidUri').and.returnValue(true);
+      spyOn(window, 'open');
+
+      expect(component.isLinkExternal(uriExpected)).not.toEqual('');
+      expect(window.open).toHaveBeenCalledWith(uriExpected, '_blank');
+      expect(component.isValidUri).toHaveBeenCalled();
```

Comment:
This expect should be called first. So, if it is not called, the test fails just there :)

Reply:
Which of the 3 goes first the one of the isValidUri

Reply:
This `expect(component.isValidUri).toHaveBeenCalled()`